src/main/java/com/hypherionmc/modpublisher/plugin/ModPublisherGradleExtension.java

```diff
@@ -226,12 +228,19 @@ public void setLoaders(ModLoader... loader) {
     }
 
     /**
-     * Add supported Java Versions for Curseforge
-     * @param version Version or Versions to add. For example: setJavaVersions("Java 8", "Java 11")
+     * Add supported Java Versions, currently only used for Curseforge.
+     * <p>
+     * Accepts any object that can be converted to a {@link JavaVersion} using {@link JavaVersion#toVersion(Object)},
+     * strings will trimmed using {@link StringUtils#substringAfterLast(String, String) substringAfterLast(version, " ")}
+     * to enable support for versions formatted as {@code "Java 8"}.
+     *
+     * @param version Version or Versions to add. For example: {@code setJavaVersions("Java 8", "11", JavaVersion.VERSION_17)}
      */
-    public void setJavaVersions(String... version) {
-        for (String jv : version) {
-            this.javaVersions.add(jv);
+    public void setJavaVersions(Object... version) {
+        for (Object v : version) {
+            // Handle strings where the java version is after a space, e.g. "Java 8"
+            if (v instanceof String) v = StringUtils.substringAfterLast((String) v, " ");
```

Comment:
Would it be better to strip the `"Java "` prefix, rather than take the substring after last `" "`?

Reply:
I think so yeah. That might be a "safer" option

Reply:
Done